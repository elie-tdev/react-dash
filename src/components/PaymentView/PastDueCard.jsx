import React from 'react';
import cn from 'classnames';
import { Box, Card, CardContent, makeStyles, Typography } from '@material-ui/core';
import InfoIcon from '@material-ui/icons/Info';
import { useTranslation } from 'react-i18next';

import PaymentLabelValue from './PaymentLabelValue';

import useGlobalStyles from '@/themes/useGlobalStyles';
import { PaymentPastDue } from '@/components';

const useStyles = makeStyles(() => ({
  root: {
    marginBottom: '4px',
  },
  headerRow: {
    marginBottom: '16px',
  },
  icon: {
    marginLeft: '0',
  },
  iconLabel: {
    display: 'flex',
    flex: 1,
    alignItems: 'center',
  },
  textBox: {
    marginTop: '16px',
  },
}));

const PastDueCard = ({ totalCurrentDueBalance, totalPastDueBalance, daysPastDue }) => {
  const globalStyles = useGlobalStyles();
  const styles = useStyles();
  const { t } = useTranslation();

  return (
    <Card data-testid="PastDueCard" className={cn(globalStyles.card, styles.root)} elevation={2}>
      <CardContent className={cn(globalStyles.cardContent)}>
        <Box className={cn(globalStyles.row, styles.headerRow, styles.iconLabel)}>
          <InfoIcon className={cn(globalStyles.icon, styles.icon, globalStyles.errorText)} />
          <Typography variant="h5" noWrap>
            {t('content.Payment Past Due')}
          </Typography>
        </Box>
        <Box className={cn(globalStyles.row)}>
          {PaymentLabelValue({
            paymentType: 'pastDue',
            paymentValue: totalPastDueBalance,
            paymentDate: totalCurrentDueBalance,
          })}
        </Box>
        <Box className={cn(styles.textBox)}>
          <PaymentPastDue pastDueDays={daysPastDue} />
        </Box>
      </CardContent>
    </Card>
  );
};

export default PastDueCard;
