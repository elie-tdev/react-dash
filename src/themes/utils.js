export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
