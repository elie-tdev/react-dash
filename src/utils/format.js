const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
