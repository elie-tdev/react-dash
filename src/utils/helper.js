const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
