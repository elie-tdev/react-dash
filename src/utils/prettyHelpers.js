export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
