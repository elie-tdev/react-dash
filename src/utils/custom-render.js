import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
