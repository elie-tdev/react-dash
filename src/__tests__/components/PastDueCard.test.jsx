import React from 'react';
import TimeAgo from 'javascript-time-ago';

import { render } from '@/utils/custom-render';
import PastDueCard from '@/components/PaymentView/PastDueCard';

describe('PastDueCard Component', () => {
  test('renders UpcomingPayment on Payment History Screen', () => {
    const { container } = render(
      <PastDueCard
        totalCurrentDueBalance="1000"
        totalPastDueBalance={500}
        daysPastDue={new TimeAgo('en-US').format(Date.now() + 3 * 24 * 60 * 60 * 1000)}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
