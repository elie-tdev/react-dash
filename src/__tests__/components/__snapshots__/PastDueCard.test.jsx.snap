// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastDueCard Component renders UpcomingPayment on Payment History Screen 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-4 makeStyles-root-30 MuiPaper-elevation2 MuiPaper-rounded"
    data-testid="PastDueCard"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent-5"
    >
      <div
        class="MuiBox-root MuiBox-root-46 makeStyles-row-8 makeStyles-headerRow-31 makeStyles-iconLabel-33"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-11 makeStyles-icon-32 makeStyles-errorText-2"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
        >
          Payment Past Due
        </h5>
      </div>
      <div
        class="MuiBox-root MuiBox-root-47 makeStyles-row-8"
      >
        <div
          class="MuiBox-root MuiBox-root-48 makeStyles-row-8 makeStyles-row-44"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-4 makeStyles-pastDueCard-43 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root makeStyles-row-8"
            >
              <div
                class="MuiBox-root MuiBox-root-54 makeStyles-container-52"
                data-testid="LabelValueColumn"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  Past Due
                </h6>
                <p
                  class="MuiTypography-root makeStyles-value-49 makeStyles-value-53 makeStyles-valueLarge-50 MuiTypography-body1"
                >
                  $500.00
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-4 makeStyles-pastDueCard-43 makeStyles-grayBackground-45 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root makeStyles-row-8"
            >
              <div
                class="MuiBox-root MuiBox-root-67 makeStyles-container-52"
                data-testid="LabelValueColumn"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6"
                >
                  Amount Due
                </h6>
                <p
                  class="MuiTypography-root makeStyles-value-49 makeStyles-value-66 makeStyles-valueLarge-50 MuiTypography-body1"
                >
                  1000
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-68 makeStyles-textBox-34"
      >
        <div
          class="MuiBox-root MuiBox-root-74 makeStyles-row-69"
        >
          <h5
            class="MuiTypography-root makeStyles-infoTitle-71 MuiTypography-h5"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-70 makeStyles-errorText-2"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            in 3 days days past due
          </h5>
          <p
            class="MuiTypography-root makeStyles-caption-7 MuiTypography-body1"
          >
            Schedule a payment for the total amount due to resolve this issue.
          </p>
        </div>
        <div
          class="MuiBox-root MuiBox-root-75 makeStyles-row-69"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-76 makeStyles-root-77 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
            data-testid="DangerButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Pay Past Due Amount
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-106"
          style="opacity: 0; visibility: hidden;"
        />
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-110"
          style="opacity: 0; visibility: hidden;"
        />
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-122"
          style="opacity: 0; visibility: hidden;"
        />
        <div
          aria-hidden="true"
          class="MuiDrawer-root MuiDrawer-modal makeStyles-confirmPanelRoot-133"
          role="presentation"
          style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
        >
          <div
            aria-hidden="true"
            class="MuiBackdrop-root"
            style="opacity: 0; visibility: hidden;"
          />
          <div
            data-test="sentinelStart"
            tabindex="0"
          />
          <div
            class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom makeStyles-modalPaper-29 MuiPaper-elevation16"
            style="visibility: hidden; webkit-transform: translateY(768px) translateY(0px); transform: translateY(768px) translateY(0px);"
            tabindex="-1"
          >
            <div
              class="MuiBox-root MuiBox-root-142 makeStyles-header-138 makeStyles-row-8"
            >
              <div
                class="MuiBox-root MuiBox-root-143 makeStyles-arrowBackBox-140"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </div>
              <p
                class="MuiTypography-root makeStyles-headerTitle-139 MuiTypography-body1"
              >
                New Payment Account
              </p>
              <div
                class="MuiBox-root MuiBox-root-144"
              />
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiDialogContent-root"
            >
              <div
                class="MuiFormControl-root makeStyles-inputField-129 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-130 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
                  data-shrink="true"
                >
                  Routing Number
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-131 MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
                  variant="outlined"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="routingNum"
                    placeholder="Routing Number"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-147"
                    >
                      <span>
                        Routing Number
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root makeStyles-inputField-129 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-130 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                  data-shrink="true"
                >
                  Account Number
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-131 MuiInputBase-fullWidth MuiInputBase-formControl"
                  variant="outlined"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="accountNum"
                    placeholder="Account Number"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-147"
                    >
                      <span>
                        Account Number
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root makeStyles-inputField-129 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-130 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                  data-shrink="true"
                >
                  Verify Account Number
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-131 MuiInputBase-fullWidth MuiInputBase-formControl"
                  variant="outlined"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="accountNumVerify"
                    placeholder="Verify Account Number"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-147"
                    >
                      <span>
                        Verify Account Number
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root makeStyles-inputFieldLast-134 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputFieldLastLabel-135 makeStyles-inputLabel-130 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                  data-shrink="true"
                >
                  Account Type
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-145 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-146"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiPaper-root MuiCard-root makeStyles-saveAsDefaultCard-136 MuiPaper-elevation0 MuiPaper-rounded"
              >
                <div
                  class="MuiBox-root MuiBox-root-149 makeStyles-saveAsDefaultBox-137"
                >
                  <p
                    class="MuiTypography-root makeStyles-inputLabel-130 MuiTypography-body1"
                  >
                    Save as default payment account
                  </p>
                  <span
                    class="MuiSwitch-root"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-150 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-151 Mui-checked"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          checked=""
                          class="PrivateSwitchBase-input-153 MuiSwitch-input"
                          type="checkbox"
                          value=""
                        />
                        <span
                          class="MuiSwitch-thumb"
                        />
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track"
                    />
                  </span>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-154"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root makeStyles-root-155 makeStyles-root-156 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
                  data-testid="PrimaryButton"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Add New Account
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            data-test="sentinelEnd"
            tabindex="0"
          />
        </div>
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-160"
          style="opacity: 0; visibility: hidden;"
        />
      </div>
    </div>
  </div>
</div>
`;
